src/core_plugins/kbn_vislib_vis_types/public/controls/heatmap_options.js

```diff
@@ -13,7 +13,7 @@ module.directive('heatmapOptions', function () {
       $scope.showLabels = false;
       $scope.customColors = false;
       $scope.options = {
-        rotateLabels: false
+        rotateLabels: $scope.vis.params.valueAxes[0].labels.rotate === 270
```

Comment:
This is a little voodoo in isolation. Consider extracting the 270 to a constant `defaultRotation` and reuse it when you are setting the default as well.